packages/easy-testing/src/DataProvider/StaticFixtureFinder.php

```diff
@@ -18,55 +17,61 @@ final class StaticFixtureFinder
 {
     public static function yieldDirectory(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $fileInfos = self::findFilesInDirectory($directory, $suffix);
-        foreach ($fileInfos as $fileInfo) {
+        $finder = self::findFilesInDirectory($directory, $suffix);
+        foreach ($finder as $fileInfo) {
             yield [new SmartFileInfo($fileInfo->getRealPath())];
         }
     }
 
     public static function yieldDirectoryExclusively(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $fileInfos = self::findFilesInDirectoryExclusively($directory, $suffix);
-        foreach ($fileInfos as $fileInfo) {
+        $finder = self::findFilesInDirectoryExclusively($directory, $suffix);
+        foreach ($finder as $fileInfo) {
             yield [new SmartFileInfo($fileInfo->getRealPath())];
         }
     }
 
-    /**
-     * @return SplFileInfo[]
-     */
-    private static function findFilesInDirectory(string $directory, string $suffix): array
+    public static function yieldDirectoryByFileName(string $directory, string $suffix = '*.php.inc'): Iterator
     {
-        $finder = Finder::create()->in($directory)->files()->name($suffix);
-        $fileInfos = iterator_to_array($finder);
+        $finder = self::findFilesInDirectory($directory, $suffix);
+        foreach ($finder as $fileInfo) {
+            yield $fileInfo->getRelativePathname() => [new SmartFileInfo($fileInfo->getRealPath())];
+        }
+    }
+
+    public static function yieldDirectoryExclusivelyByFileName(string $directory, string $suffix = '*.php.inc'): Iterator
+    {
+        $finder = self::findFilesInDirectoryExclusively($directory, $suffix);
+        foreach ($finder as $fileInfo) {
+            yield $fileInfo->getRelativePathname() => [new SmartFileInfo($fileInfo->getRealPath())];
```

Comment:
The SmartFileInfo should be used instead, for consistency with the rest of Symplify code base:

```php
$smartFileInfo = new SmartFileInfo($fileInfo->getRealPath());
yield [$smartFileInfo->getRelativePathname() => [$smartFileInfo]];
```

Reply:
By making this change, the relative path is not the same anymore.
Not sure why they work differently here.

Reply:
That's expected, as finder provides false relative paths.